cmake/configs/arm64/darwin/config.cmake

```diff
@@ -0,0 +1,19 @@
+include(CheckCCompilerFlag)
+
+set(WT_ARCH "arm64" CACHE STRING "")
```

Comment:
I'd prefer we use `aarch64` and move this file under the path `cmake/configs/aarch64/darwin/config.cmake`. Its a bit odd, but GNU uses the term `aarch64` and MacOS uses `arm64`. 
I'd prefer we use a single name to refer to the same architecture, and we've been using the `aarch64` to map to [`WT_AARCH64`](https://github.com/wiredtiger/wiredtiger/blob/0e8c23047f6b1ce08cf73f9e179270f39e1b59e8/cmake/configs/base.cmake#L11), which is subsequently used to detect our arm specific sources (i.e. in our [helper](https://github.com/wiredtiger/wiredtiger/blob/0e8c23047f6b1ce08cf73f9e179270f39e1b59e8/cmake/helpers.cmake#L599-L600) files for instance). 
Its likely we are getting arm64 from the`CMAKE_HOST_SYSTEM_PROCESSOR` variable (which is taken from the OS), but we can easily alias this in the top level CMakeLists.txt.
https://github.com/wiredtiger/wiredtiger/blob/0e8c23047f6b1ce08cf73f9e179270f39e1b59e8/CMakeLists.txt#L22-L27
Similar to our x86 builds we can force `WT_ARCH`  to `aarch64` i.e. 
```
elseif("${CMAKE_HOST_SYSTEM_PROCESSOR}" MATCHES "^(arm64|aarch64)$")
```

Reply:
Alternatively we can rename `aarch64` to `arm64` if you think that looks nicer (but would require more changes in doing so)

Reply:
Thanks, fixed in 8f64fb7. I spent a little time playing with them both and couldn't spot the difference, so that explains why.